websocketpp/transport/asio/endpoint.hpp

```diff
@@ -714,7 +712,7 @@ class endpoint : public config::socket_type {
      * @since 0.3.0
      */
     void start_perpetual() {
-        m_work.reset(new lib::asio::io_service::work(*m_io_service));
+        m_work.reset(new lib::asio::executor_work_guard<lib::asio::io_context::executor_type>(m_io_context->get_executor()));
```

Comment:
I actually suggest renaming the member variable as `m_work_guard` instead of `m_work`.